### PR TITLE
feat(Auth): Adding TOTP integration tests

### PR DESCRIPTION
**Pull request by harsh62**
### TOTP PR Number # 8

Depends on: 
* #3040 
* #3041 
* #3042
* #3043 
* #3044 
* #3045 
* #3046 

Changes
* Adds all the integration tests

_NOTE: The unit and integration tests will fail until all changes make it to the feature branch_
(The PR is directed towards a feature branch)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ruisebas — APPROVED** [review]
**Reminder**: We'll need to enable these new test to run in tvOS/watchOS as well.

**Review by royjit — APPROVED** [review]

File: AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/MFATests/TOTPSetupWhenUnauthenticatedTests.swift [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        // GIVEN
+
+        let username = "integTest\(UUID().uuidString)"
+        let password = "P123@\(UUID().uuidString)"
+
+        let didSucceed = try await AuthSignInHelper.signUpUser(
+            username: username,
+            password: password,
+            email: defaultTestEmail,
+            phoneNumber: "+16135550116")
```
Comment (thisisabhash) [inline AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/MFATests/TOTPSetupWhenUnauthenticatedTests.swift:86]: nit: should we remove any specific information and make this value configurable?

File: AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/MFATests/MFASignInTests.swift [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        try await super.setUp()
+        AuthSessionHelper.clearSession()
+    }
+
+    override func tearDown() async throws {
+        try await super.tearDown()
+        AuthSessionHelper.clearSession()
+    }
+
+    /// Test successful successful signIn with confirmSignInWithTOTPCode Step
```
[the remark this answers is not in the document]
Comment (harsh62) [reply AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/MFATests/MFASignInTests.swift:24]: Will update in the next PR.

File: AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/MFATests/TOTPSetupWhenUnauthenticatedTests.swift (hunk 1 above)
Comment (harsh62) [reply AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/MFATests/TOTPSetupWhenUnauthenticatedTests.swift:86]: Mmm... This like test phone number which is valid. I should probably do that.. Will create a follow up PR for it.

**Pull request merged**